pkg/issuer/acme/dns/dns.go

```diff
@@ -54,57 +54,28 @@ func (s *Solver) Present(ctx context.Context, crt *v1alpha1.Certificate, domain,
 	if err != nil {
 		return err
 	}
-	glog.V(4).Infof("Presenting DNS01 challenge for domain %q", domain)
+	glog.Infof("Presenting DNS01 challenge for domain %q", domain)
 	return slv.Present(domain, token, key)
 }
 
-func (s *Solver) Wait(ctx context.Context, crt *v1alpha1.Certificate, domain, token, key string) error {
-	slv, err := s.solverFor(crt, domain)
+func (s *Solver) Check(domain, token, key string) (bool, error) {
+	fqdn, value, ttl := util.DNS01Record(domain, key)
+	glog.Infof("Checking DNS propagation for %q using name servers: %v", domain, util.RecursiveNameservers)
+
+	ok, err := util.PreCheckDNS(fqdn, value)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	type boolErr struct {
-		bool
-		error
+	if ok {
+		glog.Infof("Waiting DNS record TTL (%ds) to allow propagation for propagation of DNS record for domain %q", ttl, fqdn)
+		time.Sleep(time.Second * time.Duration(ttl))
```

Comment:
This might block for a long time. Is this called in a go routine? Should there be a cap on time we wait here?

Reply:
:+1: this is probably best to be handled asynchronously, e.g. by updating the status field to indicate when it's okay to proceed.

I'm going to leave this optimisation for a follow up, as it's technically not a regression (we've always blocked the 'worker' at this point). But definitely one to address.